Documentation/ExtensionArchitecture/Extbase/Reference/_FileUpload/_BlogExcerpt.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyVendor\MyExtension\Domain\Model;
+
+use TYPO3\CMS\Extbase\Annotation\FileUpload;
+use TYPO3\CMS\Extbase\Domain\Model\FileReference;
+use TYPO3\CMS\Extbase\DomainObject\AbstractEntity;
+
+class Blog extends AbstractEntity
+{
+    #[FileUpload([
+        'validation' => [
+            'required' => true,
+            'maxFiles' => 1,
+            'fileSize' => ['minimum' => '0K', 'maximum' => '2M'],
+            'allowedMimeTypes' => ['image/jpeg', 'image/png'],
+        ],
+        'uploadFolder' => '1:/user_upload/files/',
+    ])]
+    protected ?FileReference $file = null;
```

Comment:
Just an internal note:
A "required" property has to be defined as nullable in extbase. That's sooo dump.